sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -359,35 +443,62 @@ def get_test_params(cls, parameter_set="default"):
 class PyTorchDataset(Dataset):
     """Dataset for use in sktime deep learning forecasters."""
 
-    def __init__(self, y, seq_len, fh=None, X=None):
-        self.y = y.values
-        self.X = X.values if X is not None else X
-        self.seq_len = seq_len
+    def __init__(self, y: pd.DataFrame, seq_len: int, fh=None, X: pd.DataFrame = None):
+        if not isinstance(y.index, pd.MultiIndex):
```

Comment:
Is it the same code as in the _predict method? 

Please check and if possible try to extract it to minimize code duplication